### PR TITLE
PyTorch2 Initial Experiments

### PR DESCRIPTION
**Pull request by devchandan**
TODO

- [ ] Separate training cell codes
- [ ] Add benchmarking code for batch training and non-compiled model

### CODE REVIEW

File: train_simple_gpt_v9.ipynb [hunk 1]
```diff
@@ -0,0 +1,5192 @@
...
+        "  GPU_NAME = gpu_name.replace(\" \", \"_\") # remove underscores for easier saving\n",
+        "  print(f'GPU name: {GPU_NAME}')\n",
+        "\n",
+        "  # Get GPU capability score\n",
+        "  GPU_SCORE = t.cuda.get_device_capability()\n",
+        "  print(f\"GPU capability score: {GPU_SCORE}\")\n",
+        "  if GPU_SCORE >= (8, 0):\n",
+        "    print(f\"GPU score higher than or equal to (8, 0), PyTorch 2.x speedup features available.\")\n",
+        "  else:\n",
+        "    print(f\"GPU score lower than (8, 0), PyTorch 2.x speedup features will be limited (PyTorch 2.x speedups happen most on newer GPUs).\")\n",
```
Comment (rohilverma) [inline train_simple_gpt_v9.ipynb]: Interesting. I understand GPU score is equivalent to versioning. Do we know what features are present on newer GPUs that Pytorch 2.x takes advantage of?

File: train_simple_gpt_v9.ipynb [hunk 2]
```diff
@@ -0,0 +1,5192 @@
+{
```
Comment (rohilverma) [inline train_simple_gpt_v9.ipynb]: Move into new folder `experimental/`.
Rename to `pytorch_2_experiment.ipynb`

File: train_simple_gpt_v9.ipynb (hunk 1 above)
Comment (devchandan) [reply train_simple_gpt_v9.ipynb]: I don't have a great answer to this at the moment. Been referring to documentation and started with this https://colab.research.google.com/github/mrdbourke/pytorch-deep-learning/blob/main/extras/pytorch_2_intro.ipynb which suggests we need better GPUs.

**Review by rohilverma — APPROVED** [review]

**Comment by rohilverma** [comment]
Changes look good to merge.

**Pull request merged**